Enable easier testing of the example in CI

dgreen-arm opened the pull request:
This PR has a few changes to make testing easier in CI:

1. Move the test into a subfolder. This follows the existing structure for examples.

2. Moves the locked zone printing to the start of the example. This means all the printing of information happens before any asserts, which helps with debugging.

3. Make updating Mbed TLS and Crypto in update-crypto.sh unconditional. This means we don't have to worry about whether they have already been checked out or not in the CI job before running update-crypto.sh

dgreen-arm committed:
Move atecc608a test into folder

Having the tests in a folder follows the existing structure for
examples and makes running the tests in CI simpler.

dgreen-arm committed:
Make updating Mbed TLS and Crypto unconditional

AndrzejKurek commented:
PR looks good, but I would suggest postponing it until https://github.com/ARMmbed/mbed-os-example-atecc608a/pull/7 is merged.

dgreen-arm commented:
I've rebased the PR to resolve the conflicts. I've also removed the commit that moved the locked zone printing as that's done in #7. There shouldn't be any conflicts between this and #7, so it can be merged afterwards.
